lib/buffer.h

```diff
@@ -11,9 +11,15 @@
 extern "C" {
 #endif
 
-/* Create a new buffer.  Memory will be allocated in chunks of the given
-   size.  If the argument is 0, the library will supply a reasonable
-   default size suitable for buffering socket I/O. */
+/* Default buffer size (used if none specified).  It is rounded up to the
+ * next page boundary.
+ */
+#define BUFFER_SIZE_DEFAULT 4096
```

Comment:
I don't think it's great to move this value to the header and then use it to drive several other things (including ... to square it!)
I think this should be left as it was; go ahead and add one or more tuneables to drive the new decision-making in the vty code.